### PR TITLE
metlino_sayma_ttl: Example of Metlino-master, Sayma-satellite system

### PR DESCRIPTION
**Pull request by HarryMakes**
# ARTIQ Pull Request

## Description of Changes

This adds Sayma's gateware support as a satellite controlling 2 DIO cards via an FMC-VHDCI-LVDS adaptor. This is demonstrated with an example containing a Metlino master and Sayma satellite.

## Details

The modification of Sayma RTL is to import the existing RTL library for the FMC-VHDCI-LVDS. A minor error in the library has been fixed.

The additional ARTIQ example was created based on `artiq_sinara_tester`. However, it is worth noting that certain functions originally from Sinara tester that does not run in the kernel need to be run in the kernel, as per testing. This might be due to how the list of LED and TTL devices gets initialised in my script, which might have differences from Sinara tester which I have overlooked. Nevertheless, this example suffice the requirement of demonstrating the TTL outputs in a master-satellite topology between Metlino and Sayma.

### CODE REVIEW

File: artiq/gateware/fmcdio_vhdci_eem.py [hunk 1]
```diff
@@ -19,8 +19,9 @@ def _get_connectors():
         for j, pair in enumerate(eem_fmc_connections[i]):
             for pn in "n", "p":
                 cc = "cc_" if j == 0 else ""
```
Comment (sbourdeauducq) [inline artiq/gateware/fmcdio_vhdci_eem.py:21]: This variable is no longer used, remove

File: artiq/gateware/targets/sayma_amc.py [hunk 2]
```diff
@@ -284,7 +286,7 @@ def __init__(self, platform, sys_crg, jesd_crg, dac):
 
 class Satellite(SatelliteBase):
     """
-    DRTIO satellite with local DAC/SAWG channels.
+    DRTIO satellite with local DAC/SAWG channels, as well as FMC channels.
```
Comment (sbourdeauducq) [inline artiq/gateware/targets/sayma_amc.py:289]: TTL channels via FMC and VHDCI carrier

File: artiq/gateware/targets/sayma_amc.py [hunk 3]
```diff
@@ -307,7 +309,7 @@ def __init__(self, jdcg_type, **kwargs):
         self.csr_devices.append("slave_fpga_cfg")
         self.config["SLAVE_FPGA_GATEWARE"] = 0x200000
 
-        rtio_channels = []
+        self.rtio_channels = rtio_channels = []
```
Comment (sbourdeauducq) [inline artiq/gateware/targets/sayma_amc.py:312]: Where is this used?

File: artiq/gateware/fmcdio_vhdci_eem.py (hunk 1 above)
Comment (HarryMakes) [reply artiq/gateware/fmcdio_vhdci_eem.py:21]: It is used for constructing the `eem?` connectors that the `add_std()` functions for each class in `artiq.gateware.eem` can use when requesting.

File: artiq/gateware/targets/sayma_amc.py (hunk 3 above)
Comment (HarryMakes) [reply artiq/gateware/targets/sayma_amc.py:312]: Used by `add_std()` functions in `artiq.gateware.eem`

File: artiq/gateware/fmcdio_vhdci_eem.py (hunk 1 above)
Comment (sbourdeauducq) [reply artiq/gateware/fmcdio_vhdci_eem.py:21]: OK sorry I missed it

**Pull request merged**